File: Daily Challanges/2026-05-May/2026-05-23.js
function getOpenIssues(issues, prs) {
  function rotations(s) {
    const result = new Set();
    for (let i = 0; i < s.length; i++) {
      result.add(s.slice(i) + s.slice(0, i));
    }
    return result;
  }
  function closes(issue, pr) {
    let a = String(issue);
    let b = String(pr);
    const n = Math.max(a.length, b.length);

    a = a.padStart(n, "0");
    b = b.padStart(n, "0");

    if (a === b) return false;
    return rotations(a).has(b);
  }
  const openIssues = [];
  for (const issue of issues) {
    let closed = false;

    for (const pr of prs) {
      if (closes(issue, pr)) {
        closed = true;
        break;
      }
    }
    if (!closed) openIssues.push(issue);
  }
  return openIssues;
}

console.log(getOpenIssues([123, 234], [231]));
console.log(getOpenIssues([123, 345, 16], [345, 231]));

File: Daily Challanges/2026-05-May/2026-05-23.py
def get_open_issues(issues, prs):
    def rotations(s):
        return {s[i:] + s[:i] for i in range(len(s))}
    def closes(issue, pr):
        a = str(issue)
        b = str(pr)
        n = max(len(a), len(b))
        a = a.zfill(n)
        b = b.zfill(n)
        if a == b:
            return False
        return b in rotations(a)
    open_issues = []
    for issue in issues:
        closed = False
        for pr in prs:
            if closes(issue, pr):
                closed = True
                break
        if not closed:
            open_issues.append(issue)
    return open_issues


print(get_open_issues([123, 234], [231]))
print(get_open_issues([123, 345, 16], [345, 231]))